external/fv3fit/tests/training/test_train.py

```diff
@@ -59,7 +60,7 @@ def assert_can_learn_identity(
     model, test_dataset = train_identity_model(
         model_type, sample_func=sample_func, hyperparameters=hyperparameters
     )
-    out_dataset = model.predict(test_dataset)
+    out_dataset = model.predict(stack_non_vertical(test_dataset)).unstack("sample")
```

Comment:
Similar question about predict, it seems like we should test two modes - stacked to stacked, and unstacked to unstacked. Thorough tests only need to be done for unstacked to unstacked.